modules/FRDRAPIClient.py

```diff
@@ -0,0 +1,134 @@
+                pub_tokens['access_token'], pub_tokens['expires_at_seconds'])
+
+            pub_client = DataPublicationClient(
+                base_url, authorizer=pub_authorizer)
+            self._pub_client = pub_client
+        except Exception as e:
+            self._logger.error(
+                "Failed to auth for FRDR API usage. {}".format(e))
+            raise Exception(e)
+
+    def update_requestitem(self, data):
+        """Update requestitem data on FRDR when depositors grant access
+           to the key on FRDR Encryption App.
+
+        Args:
+            data (dict): {"expires": The expiry data of the granted access, 
+                          "vault_dataset_id": The id for the dataset on Vault,
+                          "vault_requester_id": The id for the requester on Vault}
+
+        Returns:
+            [string]: REST API call response 
+        """
+        return self._pub_client.update_requestitem(data)
+
+    def _load_auth_client(self):
+        return NativeAppAuthClient(
+            config.GLOBUS_DATA_PUBLICATION_CLIENT_ID,
+            app_name=config.GLOBUS_DATA_PUBLICATION_CLIENT_NAME)
+
+    def _token_response_to_dict(self, token_response):
+        resource_servers = ('publish.api.frdr.ca', )
+        ret_toks = {}
+
+        for res_server in resource_servers:
+            try:
+                token = token_response.by_resource_server[res_server]
+                ret_toks[res_server] = token
+            except:
+                pass
+
+        return ret_toks
+
+    def _interactive_login(self):
+        native_client = self._load_auth_client()
+
+        port = Util.find_free_port()
+        native_client.oauth2_start_flow(
+            requested_scopes=(
+                'urn:globus:auth:scope:publish.api.frdr.org:all'),
+            refresh_tokens=True,
+            redirect_uri='http://localhost:{}'.format(port))
+        auth_url = native_client.oauth2_get_authorize_url()
+        webbrowser.open(auth_url)
+        auth_code = self._login_get_token(port)
+
+        if auth_code is None:
+            raise TimeoutError(
+                "You login process has expired. Please try again")
+
+        tkns = native_client.oauth2_exchange_code_for_tokens(auth_code)
+        return self._token_response_to_dict(tkns)
+
+    def _login_get_token(self, port):
+
+        class HttpServ(HTTPServer):
+            def __init__(self, *args, **kwargs):
+                HTTPServer.__init__(self, *args, **kwargs)
+                self.token = None
+
+        class AuthHandler(BaseHTTPRequestHandler):
+            token = ''
+
+            def do_GET(self):
+                params = parse.parse_qs(self.path.split('?')[1])
+                self.server.token = params['code'][0]
+                self.send_response(200)
+                self.end_headers()
+                self.wfile.write(str.encode(
+                    '<div>Authentication successful, you can close the browser now.</div>'))
```

Comment:
Translation possible at this point?

Reply:
Done in https://github.com/frdr-dfdr/frdr_encryption_client/pull/1/commits/a09232f1c01b1d624454ed8061103c16fab9e54b